contracts/did-stellar-registry/src/contract.rs

```
@@ -0,0 +1,394 @@
+
+fn validate_service(e: &Env, s: &DidService) {
+    if s.id_suffix.len() > MAX_SERVICE_ID_LEN {
+        panic_with_error!(e, RegistryError::ServiceIdTooLong);
+    }
+    if s.id_suffix.len() == 0 {
+        panic_with_error!(e, RegistryError::ServiceIdInvalidFormat);
+    }
+    if !is_valid_id_suffix(&s.id_suffix) {
+        panic_with_error!(e, RegistryError::ServiceIdInvalidFormat);
+    }
+    if s.service_type.len() == 0 || s.service_type.len() > MAX_SERVICE_TYPE_LEN {
```

Clippy (1.95.0)

[warning] length comparison to zero

(warning)